storage.bs

```diff
@@ -203,6 +203,16 @@ anticipated that some APIs will be applicable to both <a>storage types</a> going
 <p class=XXX>This is expected to change; see
 <a href="https://privacycg.github.io/storage-partitioning/">Client-Side Storage Partitioning</a>.
 
+<p>To <dfn export>obtain a storage key for non-storage purposes</dfn>, given an <a>environment
+settings object</a> <var>environment</var>, run these steps:
+
+<ol>
+ <li><p>Let <var>key</var> be <var>environment</var>'s
+ <a for="environment settings object">origin</a>.
```

Comment:
You forgot to remove this step from the algorithm below.

Reply:
We still need the step there for the opaque origin check, but I'll change the variable name since 'key' there is no longer accurate

Reply:
I see, we might need a bigger change.

In the future key will be at least (site, origin) + implementation-defined fields. Now if site is an opaque origin the algorithm below will need to return failure as well. So I think it does want to operate on the eventual key.

Reply:
> Now if site is an opaque origin the algorithm below will need to return failure as well.

Off-topic for this PR, but I don't think we (Chrome) are currently planning to block storage in an embedded iframe merely because the top-level site has an opaque origin (i.e. as long as origin isn't opaque, the opaqueness of the site shouldn't matter).

Reply:
> In the future key will be at least (site, origin) + implementation-defined fields. Now if site is an opaque origin the algorithm below will need to return failure as well. So I think it does want to operate on the eventual key.

The `environment` object has the [`top-level-origin`](https://html.spec.whatwg.org/multipage/webappapis.html#concept-environment-top-level-origin) field, so for determining whether the top-level origin is opaque it might be easier to use that (in either algorithm, as appropriate) than to define a way to extract `site` from the storage key (IIUC).

Reply:
I think we could just check if either key[0] or key[1] is an opaque origin, assuming we make it a tuple and those are the first two fields.

Reply:
@mkruisselbrink so you'd have some kind of ephemeral storage?

Reply:
> @mkruisselbrink so you'd have some kind of ephemeral storage?

Yeah, behavior (and implementation) wise this would be pretty similar to storage in anonymous iframes (If I understand the anonymous iframe proposal correctly, the desire for anonymous iframes to be able to access (ephemeral) storage is one reason they don't want the iframes to have opaque origins).

Reply:
The latest commit turns storage key into a tuple (with only one item for now) and uses its `origin` for the opaque origin comparison. Per the [tuple docs](https://infra.spec.whatwg.org/#tuples) we can avoid using the indexing syntax if we name the tuple items, so I went with that approach.

Also, regarding https://github.com/whatwg/html/pull/7567#discussion_r801790301, I'll plan to add a storage key comparison function in a separate PR unless you all prefer that it be a part of this one.